### PR TITLE
Update Data SDK for C++ documentation

### PR DESCRIPTION
**Pull request by nplakhtii**
Update documentation due to technical writing standards.

Moved the Installation instructions from README.md to GettingStartedGuide.md.
Moved the "Build the SDK" section from README.md to GettingStartedGuide.md.
Moved the "Why Use" topic to Introduction (in GitLab).
Added Table of Contents.
Added the reference link in README.
Changed outdated platform URLs to actual ones.

Relates-To: TECHDOCS-1227

Signed-off-by: Nataliia Plakhtii ext-nataliia.plakhtii@here.com

### CODE REVIEW

**Comment by nplakhtii** [comment]
@andrey-kashcheev 
I made a pull request into the TECHDOCS-1227-update branch to add some changes.

I can now see your changes (https://github.com/heremaps/here-data-sdk-cpp/pull/1347) in the current pull request.

Could you please suggest any additional actions to be taken to not to have your changes merged here?

**Comment by nplakhtii** [comment]
> @andrey-kashcheev I made a pull request into the TECHDOCS-1227-update branch to add some changes.
> 
> I can now see your changes (#1347) in the current pull request.
> 
> Could you please suggest any additional actions to be taken to not to have your changes merged here?

Fixed.

File: docs/CMakeLists.txt [hunk 1]
```diff
@@ -40,8 +40,8 @@ set(COMPONENT_DIRS
 )
 set(MAINPAGE_FILE README.md)
 set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ${MAINPAGE_FILE})
+set(PREDEFINED OLP_SDK_ENABLE_DEFAULT_CACHE)
```
Comment (andescu) [inline docs/CMakeLists.txt:43]: OLP_SDK_ENABLE_DEFAULT_CACHE is something totally different. Are you sure this is correct what you are doing or simply a copy-paste-error?

File: docs/GettingStartedGuide.md [hunk 2]
```diff
@@ -38,7 +53,11 @@ You can use the Data SDK in your CMake project or install it on your system.
 
 **To install the Data SDK on your system:**
 
+<<<<<<< HEAD
```
Comment (andescu) [inline docs/GettingStartedGuide.md:56]: These are git merge conflict markers. Please fix them first and then push.

File: docs/CMakeLists.txt (hunk 1 above)
Comment (nplakhtii) [reply docs/CMakeLists.txt:43]: Copy-paste error. Definitely. Already checked in the source files

File: README.md [hunk 3]
```diff
@@ -37,7 +37,7 @@ For more information on Data API, see its <a href="https://developer.here.com/do
 
 When new API is introduced in Data SDK for C++, the old one is not deleted straight away. The standard API deprecation time is 6 months. It gives you time to switch to new code. However, we do not provide ABI backward compatibility.
 
-All of the deprecated methods, functions, and parameters are documented in the Data SDK for C++ <a href="https://developer.here.com/documentation/sdk-cpp/api_reference/index.html" target="_blank">API Reference</a> and <a href="https://github.com/heremaps/here-data-sdk-cpp/blob/master/CHANGELOG.md" target="_blank">changelog</a>.
+Learn more about deprecated methods, functions, and parameters in <a href="https://github.com/heremaps/here-data-sdk-cpp/blob/master/CHANGELOG.md" target="_blank">changelog</a>.
```
Comment (HalynaDumych) [inline README.md:40]: Why did you remove the link to the API reference?
It is where users can check whether the needed methods, functions, and parameters are deprecated.

File: README.md [hunk 4]
```diff
@@ -69,6 +69,8 @@ The table below lists the dependencies of the Data SDK.
 | Snappy               | 1.1.7           |
 | RapidJSON            | latest          |
 
+### Additional Linux dependencies
```
Comment (HalynaDumych) [inline README.md:72]:
```suggestion
### Linux dependencies
```

File: docs/CMakeLists.txt (hunk 1 above)
Comment (HalynaDumych) [reply docs/CMakeLists.txt:43]: @nplakhtii, we agreed to first test changes to the CMake list locally and then create a CR. Therefore, this CR should not contain any changes in CMakeLists.txt. Please see this file in master (https://github.com/heremaps/here-data-sdk-cpp/blob/master/docs/CMakeLists.txt) and make sure your CR does not change it.

File: docs/GettingStartedGuide.md [hunk 5]
```diff
@@ -38,7 +53,7 @@ You can use the Data SDK in your CMake project or install it on your system.
 
 **To install the Data SDK on your system:**
 
-1. Install all the dependencies needed for the Data SDK.<br>For more information on dependencies, see the [Dependencies](#dependencies) and [Additional Linux dependencies](#additional-linux-dependencies) sections.
+1. Install all the dependencies needed for the Data SDK.<br>For more information on dependencies, see the [Dependencies](../README.md#dependencies) and [Additional Linux dependencies](../README.md#additional-linux-dependencies) sections in the README.md file.
```
Comment (HalynaDumych) [inline docs/GettingStartedGuide.md:56]: If you change the "Additional Linux dependencies" section according to the comment above, make sure to also update the link here.

**Review by HalynaDumych — CHANGES_REQUESTED** [review]

File: docs/CMakeLists.txt (hunk 1 above)
Comment (nplakhtii) [reply docs/CMakeLists.txt:43]: sorry, my bad. removing this tag (will commit in another MR).

File: README.md (hunk 3 above)
Comment (nplakhtii) [reply README.md:40]: The API Reference, which I deleted, leads us to the following link > https://developer.here.com/documentation/sdk-cpp/1.15.0/api_reference/index.html (developer.here.com > HERE Data SDK for C++ > API Reference C++. 

Now, we are in the **README.md** file, which contains same content (=sections) as in the **API Reference for C++** on developer.here.com ((https://developer.here.com/documentation/sdk-cpp/1.15.0/api_reference/index.html):
2. Test coverage
3. Backward compatibility
4. Supported platforms
5. Dependencies
6. Use the SDK
7. Contribution
8. License

Are those files different? 
I mean the **README.md** and the file on **https://developer.here.com/documentation/sdk-cpp/1.15.0/api_reference/index.html**?

File: README.md (hunk 4 above)
Comment (nplakhtii) [reply README.md:72]: Changed locally.
I will commit the change with all other implementations.

File: docs/GettingStartedGuide.md (hunk 5 above)
Comment (nplakhtii) [reply docs/GettingStartedGuide.md:56]: Changed.

File: docs/GettingStartedGuide.md [hunk 6]
```diff
@@ -159,4 +174,4 @@ HERE Data SDK for C++ contains several example programs that demonstrate some of
 - [Read example](dataservice-read-catalog-example.md) – demonstrates how to get catalog and partition metadata, as well as partition data.
 - [Read example for a stream layer](dataservice-read-from-stream-layer-example.md) – demonstrates how to get data from a stream layer.
 - [Cache example](dataservice-cache-example.md) – demonstrates how to get partition data and work with a mutable and protected cache.
-- [Write example](dataservice-write-example.md) – demonstrates how to publish data to OLP.
+- [Write example](dataservice-write-example.md) – demonstrates how to publish data to OLP.
```
[the remark this answers is not in the document]
Comment (nplakhtii) [reply docs/GettingStartedGuide.md:177]: Changed.

File: README.md (hunk 3 above)
Comment (nplakhtii) [reply README.md:40]: Discussed. Moved the "in the Data SDK for C++ <a href="https://developer.here.com/documentation/sdk-cpp/api_reference/index.html" target="_blank">API Reference</a>" link back.

File: README.md [hunk 7]
```diff
@@ -93,4 +96,5 @@ Copyright (C) 2019–2022 HERE Europe B.V.
 For license details, see the <a href="https://github.com/heremaps/here-data-sdk-cpp/blob/master/LICENSE" target="_blank">LICENSE</a> file in the root of this project.
 
 > #### Note
+
```
Comment (HalynaDumych) [inline README.md:99]: Please delete this empty line.

**Review by andescu — APPROVED** [review]

**Review by mykolamalik — APPROVED** [review]

**Review by HalynaDumych — APPROVED** [review]
